stubs/Drupal/Core/TypedData/Plugin/DataType/ItemList.stub

```diff
@@ -13,4 +13,10 @@ use Drupal\Core\TypedData\TypedDataInterface;
  * @implements ListInterface<T>
  */
 class ItemList extends TypedData implements \IteratorAggregate, ListInterface {
+
+  /**
+   * @return \ArrayIterator<int, T>
+   */
+  public function getIterator(): \ArrayIterator {}
```

Comment:
I wonder if we should split this out into its own PR, just show it is more apparent. It was crucial to fixing this PR and removes ignores errors in Drupal core's baseline. 

"Stub ItemList::getIterator return type" and then the PR can contain your findings

Reply:
No problem, I can prepare PRs with this change. Should this one: https://github.com/mglaman/phpstan-drupal/pull/846#issuecomment-2790236394 be part of new PR? Because they are kind of connected. Or should I keep it with this PR?